### DISCUSSION TITLE
Data automatically restored after a factory reset

### DISCUSSION THREAD

**Discussion by UntypedLetter** [TeamNewPipe/NewPipe#10421] [Questions] [answered]
After a factory reset, how did a freshly installed NewPipe restore all my subscriptions and history automatically? Rather automagically! :-)

**Comment by TobiGr** [comment, marked as answer]
I guess your device is linked to a Goolge account and uses Goolge Drive Backups or some other device backup service (Nextcloud, ...) which backup and restore app data automatically (even after a fresh install of the app). I think there even is an option to copy the app data on a fresh install in your device's backup settings.

**Comment by UntypedLetter** [comment]
That solves the mystery!

Thank you, Tobi! :-)
